tests/test_crawl.py

```diff
@@ -0,0 +1,287 @@
+from sh_scrapy.crawl import _launch
+from sh_scrapy.crawl import list_spiders
+from sh_scrapy.crawl import main
+from sh_scrapy.log import HubstorageLogHandler
+
+
+@mock.patch.dict(os.environ, {'HWORKER_SENTRY_DSN': 'hw-sentry-dsn',
+                              'SENTRY_DSN': 'sentry-dsn'})
+def test_init_module():
+    assert sh_scrapy.crawl._sys_stderr == sys.stderr
+    assert sh_scrapy.crawl._sys_stdout == sys.stdout
+    assert sh_scrapy.crawl.socket.getdefaulttimeout() == 60.0
+
+
+def test_ignore_warnings():
+    with ignore_warnings(category=ScrapyDeprecationWarning):
+        warnings.warn("must be suppressed", ScrapyDeprecationWarning)
+
+
+@mock.patch('traceback.print_exception')
+def test_fatal_error(trace_print):
+    exception = ValueError('some exception')
+    traceback = None
+    try:
+        raise exception
+    except:
+        # get traceback before we cleaned it with fatalerror
+        traceback = sys.exc_info()[2]
+        _fatalerror()
+    assert trace_print.called
+    trace_args = trace_print.call_args_list[0]
+    assert trace_args[0][0] == ValueError
+    assert trace_args[0][1] == exception
+    assert trace_args[0][2] == traceback
+    assert trace_args[0][3] is None
+    assert trace_args[0][4] == sys.stderr
+
+
+@mock.patch('traceback.print_exception')
+def test_fatal_error_ignore_IOError(trace_print):
+    trace_print.side_effect = IOError('some error')
+    try:
+        raise ValueError('some exception')
+    except:
+        _fatalerror()
+    assert trace_print.called
+
+
+@mock.patch('sh_scrapy.crawl._sentry_dsn')
+def test_fatal_error_sentry_import_error(sentry_dsn):
+    try:
+        raise ValueError('some exception')
+    except:
+        _fatalerror()
+
+
+@mock.patch('sh_scrapy.crawl._sentry_dsn')
+def test_fatal_error_sentry_with_mock(sentry_dsn):
+    raven_stub = type('raven', (object, ), {})
+    raven_stub.Client = mock.Mock()
+    try:
+        sys.modules['raven'] = raven_stub
+        raise ValueError('some exception')
+    except:
+        _fatalerror()
+    finally:
+        del sys.modules['raven']
+    assert raven_stub.Client.called
+    assert raven_stub.Client.call_args[0] == (sentry_dsn,)
+    sentry_client = raven_stub.Client.return_value
+    assert sentry_client.captureException.called
+
+
+@mock.patch('sh_scrapy.crawl._sentry_dsn')
+def test_fatal_error_sentry_with_mock_ignore_errors(sentry_dsn):
+    raven_stub = type('raven', (object, ), {})
+    raven_stub.Client = mock.Mock()
+    sentry_client = raven_stub.Client.return_value
+    sentry_client.captureException.side_effect = IOError('error')
+    try:
+        sys.modules['raven'] = raven_stub
+        raise ValueError('some exception')
+    except:
+        _fatalerror()
+    finally:
+        del sys.modules['raven']
+
+
+def test_get_apisettings_empty():
+    assert _get_apisettings() == {}
+
+
+@mock.patch.dict(os.environ, {
+    'SHUB_SETTINGS': 'data:;base64,ImhlbGxvIHdvcmxkIg=='})
```

Comment:
Well, because you already added these tests - let's merge this PR and remove data URI scheme support from `decode_uri` in another PR (to make this changeset related to Py3 porting only). Be careful because it's quite often imported in user code.